firedrake/slate/slate.py

```diff
@@ -242,6 +242,7 @@ class Tensor(TensorBase):
     """
 
     operands = ()
+    prec = None
```

Comment:
What is the backtrace? Perhaps you just need to add `Tensor` with the _most binding_ precedence. I suspect you are now comparing `None` to `int`s which does not work in Python 3.